src/live/customize.rs

```diff
@@ -169,6 +175,28 @@ impl LiveInitrd {
         Ok(())
     }
 
+    pub fn network_nmstate(&mut self, path: &str) -> Result<()> {
+        if !self.features.live_initrd_network {
+            bail!("This OS image does not support customizing network settings.");
+        }
+        let net_state_str = read_to_string(path)?;
+        match NetworkState::new_from_json(net_state_str.as_str()) {
+            Ok(net_state) => match net_state.gen_conf() {
+                Ok(nm_conf) => {
+                    let nm_conf_str = serde_json::to_string(&nm_conf)?;
+                    let name = filename(path)?;
+                    let path = format!("{}/{}.nmconnection", INITRD_NETWORK_DIR, name);
```

Comment:
Hmm, won't this create something like `my-config.yaml.nmconnection`?

Reply:
The final PR will just use base name so the extension is skip and we get `my-config.nmconnection`.